homeassistant/helpers/service.py

```diff
@@ -530,12 +544,12 @@ async def admin_handler(call: ha.ServiceCall) -> None:
 def verify_domain_control(hass: HomeAssistantType, domain: str) -> Callable:
     """Ensure permission to access any entity under domain in service call."""
 
-    def decorator(service_handler: Callable) -> Callable:
+    def decorator(service_handler: Callable[[ha.ServiceCall], Any]) -> Callable:
```

Comment:
```suggestion
    def decorator(service_handler: Callable[[ha.ServiceCall], None]) -> Callable:
```

Reply:
If we change this, we need to change the return type of `check_permissions` below accordingly, but that seems to cause a set of additional issues to address, e.g.
```
homeassistant/helpers/service.py:555: error: Returning Any from function declared to return "None"
homeassistant/helpers/service.py:555: error: Incompatible types in "await" (actual type "None", expected type "Awaitable[Any]")
homeassistant/helpers/service.py:586: error: Returning Any from function declared to return "None"
homeassistant/helpers/service.py:586: error: Incompatible types in "await" (actual type "None", expected type "Awaitable[Any]")
```
I haven't dug into this more closely if this is what should be done, just wanted to raise in case you already know for sure which way it should be, i.e. if the Any -> None change is valid and desirable.

Reply:
Merged without this change; we can improve it later if appropriate.